chainer/functions/activation/softmax.py

```diff
@@ -36,17 +37,17 @@ def forward(self, x):
             one = numpy.array(1, dtype=oz_dtype).ctypes
             zero = numpy.array(0, dtype=oz_dtype).ctypes
             handle = cudnn.get_handle()
-            x_cube = x[0].reshape(x[0].shape[:2] + (-1, 1))
+            x_cube = x[0].reshape(self._get_cube_shape(x[0].shape))
             desc = cudnn.create_tensor_descriptor(x_cube)
             self.y = xp.empty_like(x[0])
             libcudnn.softmaxForward(
                 handle, _algorithm, _mode, one.data, desc.value,
                 x_cube.data.ptr, zero.data, desc.value,
                 self.y.data.ptr)
         else:
```

Comment:
Thank you for looking that up. If I understand correctly, `x_cube` is reshaped so that the axis specified by the caller is always moved to the second axis (along which the computation is performed with `CUDNN_SOFTMAX_MODE_CHANNEL`), so wouldn't it work for any given axis?